### ISSUE TITLE
Feature request: web gui 

### ISSUE THREAD

**Issue by chappyj** [dcs-liberation/dcs_liberation#104]
Exploring the idea of an alternative ui web app to run dcs liberation remotely and collaboratively with other users. Multiple users could flight plan and make operational changes via the web gui, resulting miz creation could enable play on dedicated servers that are hosting the web server also

**Comment by Khopa** [comment]
I agree that this would be cool. But this would also basically mean re-starting from scratch.

I originally considering doing a web based app at some point. But this being a desktop app make it rather simple to install and use for single player.

It could a "desktop webapp", with electron or something similar. But I think you'd also have to either entirely rewrite pydcs (the library i use to generate mission file) in order to get a js library able to generate DCS mission file, OR find a way to call python pydcs code through electron/js. Or else have a separate Python server that handle the job, but that would be much overhead... Which is why i gave up on this idea and went for a straightforward Python QT app (i was also starting from shdwp original code, which made it easier to iterate).

It's true that this being a webapp would have its perks though. I don't deny it. It would be much easier to develop and customize UI, and we could use leaflet.js to display nice maps and stuff like that... 

But this would be such a big rework of the project, for me it's a bit out of scope for DCS Liberation. It would be an entirely new project.

**Comment by HomeFries94** [comment]
Here's a "mountain to Mohammed" question:

If the point is collaboration, then instead of a web interface, would it be possible to add code to "import" a save game from a client (who edits his flight) and just allow the host to choose which changes to apply?  I know it's a different type of coding altogether, and I don't know whether Python's parsing algorithms are up to the task, but it's worth asking.

**Comment by DanAlbert** [comment]
That sounds incredibly error prone. We want a new UI framework for other reasons anyway.

**Comment by HomeFries94** [comment]
No problem.  Just a brainstorm.

**Comment by Canadave-eh** [comment]
How about an interim solution of just exporting key campaign details to a specified discord channel (or in an easily postable format).
- Add button available at the end of each turn "Export to Discord" or "Export to html"
- Would package the current map, last turn results,  next turn flights, next turn briefing information.  Additional information exported based on user selections (check boxes) and what is available
- Depending how difficult it is to do - automatically push export to a user enterable discord channel and forum - or export in a format that is easily "pasted" manually into a discord channel by the user
- Could add an option to export into simple html for easy integration into an existing user controlled web site

**Comment by wartech0** [comment]
I have a very rudimentary web-gui setup on my branch I am working on using react as a front end. I plan on adding an authentication system and giving specific users authority over the game/squadron based on configuration.

**Comment by DanAlbert** [comment]
How are you going about that? We already have a react app within the Qt app. We do have a plan for this that's been in progress for a while, it's just a thing that takes years. The plan is to continue moving more of the existing UI into the part that's already react, and once we're done, jettison Qt and replace it with electron (with a configurable backend that defaults to localhost).

A mistake we need to avoid repeating though: the UI needs tests. It's historically the most error prone part of Liberation and is resistant to tools to catch even the most obvious bugs (type checking). The map (the one part that is currently react) is quite a bit better tested than the rest, and tbh I wasn't planning on any more changes to that part of the UI until coverage is closer to 100% (at least, any new components ought to have tests).

If you want to tackle this, definitely do it it incrementally like that.

**Comment by wartech0** [comment]
Right now I got a basic example working alongside the current UI but its very very rudimentary, I'll continue working on it and I'll write tests for new components as well. I'm trying to avoid anything that will affect the functionality of the current qt / map stuff.

**Comment by DanAlbert** [comment]
> I'm trying to avoid anything that will affect the functionality of the current qt / map stuff.

I think that's exactly what you need to do though :) How are you planning to land this otherwise?
